### PR TITLE
Use should_run instead of showable

### PR DESCRIPTION
**Pull request by poncovka**
Create a new object that will provide access to all persistent objects of the
user interface. The goal is to replace all these objects with this one. In the
future, the context might be replaced with a DBus module.

Define the `should_run` method instead of the `showable` property to prevent
the initialization of spokes that shouldn't run at all.

### CODE REVIEW

**Review by M4rtinK — APPROVED** [review]
Looks good to me - nice improvement! :+1: 

BTW, I guess some addons might be using the showable property and might need to be adjusted ? Initial Setup does not seem to be touching it.

File: pyanaconda/ui/context.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    @property
+    def anaconda(self) -> Anaconda:
+        """The Anaconda object."""
+        return self._anaconda
+
+    @property
+    def flags(self) -> Flags:
+        """The Anaconda flags."""
+        return flags
```
Comment (jkonecny12) [inline pyanaconda/ui/context.py:47]: I don't think you need this property here. Let's remove it because it's not used anywhere.

**Review by jkonecny12 — APPROVED** [review]
Looks good to me otherwise. Although, I would recommend to run complete KS tests suite to this to have at least some check that it's working as expected.

EDIT:
Of course if you need it fixed soon then not wait for KS tests.

**Review by rvykydal — APPROVED** [review]
Looks good to me.

File: pyanaconda/ui/context.py (hunk 1 above)
Comment (poncovka) [reply pyanaconda/ui/context.py]: I have removed the flags for now.

**Comment by poncovka** [comment]
The failing coverage tests are not related to this issue.

**Pull request merged**